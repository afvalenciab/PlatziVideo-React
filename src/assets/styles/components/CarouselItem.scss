.carrousel__item {
  border-radius: 20px;
  overflow: hidden;
  position: relative;
  transition: 0.3s all;
  transform-origin: center left;
  cursor: pointer;

  figure {
    margin: 0;
    img {
      vertical-align: top;
      display: inline-block;
      width: 200px;
      height: 250px;
      object-fit: cover;
    }
  }
}

.carrousel__list .carrousel__item:hover {
  transform: scale(1.4);
  opacity: 1;
}
.carrousel__item:hover ~ .carrousel__item {
  transform: translateX(80px);
}
.carrousel__item:hover .carrousel__item-cover {
  display: block;
}

.carrousel__item-cover {
  position: absolute;
  width: 200px;
  bottom: 0;
  padding: 10px;
  box-sizing: border-box;
  background: linear-gradient(transparent, rgba(0, 0, 0, 0.9));
  display: none;

  p {
    margin: 0;
  }

  .tittle__movie {
    font-size: 16px;
    font-weight: 600px;
  }

  .detail__movie {
    font-size: 12px;
  }
}

.icon {
  display: inline-block;
  color: #21c08b;
  font-size: 18px;
  font-weight: bold;
  border-radius: 50%;
  background: #fff;
  margin-right: 5px;
}

.icon-plus:before {
  content: '\e902';
}

.icon-play:before {
  content: '\e903';
}

.icon-remove:before {
  content: "\e904";
  display: inline-block;
  color: #fff;
  font-size: 18px;
  font-weight: bold;
  margin-right: 5px;
}
