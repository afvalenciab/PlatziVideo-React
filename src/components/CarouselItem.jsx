import React from 'react';
import PropTypes from 'prop-types';
import '../assets/styles/components/CarouselItem.scss';
import imgCover1 from '../assets/static/above-action-aerial-2346289.jpg';

const CarouselItem = ({ cover, title, year, contentRating, duration }) => {
  return (
    <div className='carrousel__item'>
      <figure>
        <img
          src={cover}
          alt={title}
        />
      </figure>
      <div className='carrousel__item-cover'>
        <i className='icon-play icon' />
        <i className='icon-plus icon' />
        <p className='title__movie'>{title}</p>
        <p className='detail__movie'>
          {`${year} ${contentRating} ${duration} minutos`}
        </p>
      </div>
    </div>
  );
};

CarouselItem.propTypes = {
  cover: PropTypes.string.isRequired,
  title: PropTypes.string,
  year: PropTypes.number,
  contentRating: PropTypes.string,
  duration: PropTypes.number,
};

export default CarouselItem;
