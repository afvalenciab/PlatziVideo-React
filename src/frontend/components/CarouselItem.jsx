/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { setFavoriteUserMovies, deleteFavorite } from '../actions';
import '../assets/styles/components/CarouselItem.scss';

const CarouselItem = (props) => {
  const { id, _id, cover, title, year, contentRating, duration, isList, user } = props;

  const handleSetFavorite = () => {
    props.setFavoriteUserMovies({
      id, _id, cover, title, year, contentRating, duration, userId: user.id,
    });
  };

  const handleDeleteFavorite = (itemId) => {
    props.deleteFavorite(itemId);
  };

  return (
    <div className='carrousel__item'>
      <figure>
        <img
          src={cover}
          alt={title}
        />
      </figure>
      <div className='carrousel__item-cover'>

        <Link to={`/player/${id}`}>
          <i className='icon-play icon' />
        </Link>

        { isList ?
          <i className='icon-remove' onClick={() => handleDeleteFavorite(id)} /> :
          <i className='icon-plus icon' onClick={handleSetFavorite} /> }
        <p className='title__movie'>{title}</p>
        <p className='detail__movie'>
          {`${year} ${contentRating} ${duration} minutos`}
        </p>
      </div>
    </div>
  );
};

CarouselItem.propTypes = {
  cover: PropTypes.string.isRequired,
  title: PropTypes.string,
  year: PropTypes.number,
  contentRating: PropTypes.string,
  duration: PropTypes.number,
};

const mapDispatchToProps = {
  setFavoriteUserMovies,
  deleteFavorite,
};

const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CarouselItem);
