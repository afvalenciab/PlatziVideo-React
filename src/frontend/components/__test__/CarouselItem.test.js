import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMocks';
import CarouselItem from '../CarouselItem';

describe('Carousel Item testing', () => {
  test('Match Snapshot', () => {
    const carousel = create(
      <ProviderMock>
        <CarouselItem />
      </ProviderMock>,
    );
    expect(carousel.toJSON()).toMatchSnapshot();
  });
});
