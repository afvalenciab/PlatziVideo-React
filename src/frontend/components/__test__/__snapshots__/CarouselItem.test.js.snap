// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Item testing Match Snapshot 1`] = `
<div
  className="carrousel__item"
>
  <figure>
    <img />
  </figure>
  <div
    className="carrousel__item-cover"
  >
    <a
      href="/player/undefined"
      onClick={[Function]}
    >
      <i
        className="icon-play icon"
      />
    </a>
    <i
      className="icon-plus icon"
      onClick={[Function]}
    />
    <p
      className="title__movie"
    />
    <p
      className="detail__movie"
    >
      undefined undefined undefined minutos
    </p>
  </div>
</div>
`;
